File: daily/128-Longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        elements = set(nums)
        maxLength = 0
        for num in elements:
            if num - 1 not in elements:
                current = 0
                while num + current in elements:
                    current += 1
                maxLength = max(maxLength, current)

        return maxLength

File: daily/Arrays/128-Longest-consecutive-sequence.go
package daily

func longestConsecutive(nums []int) int {
	numSet := make(map[int]bool)
	for _, num := range nums {
		numSet[num] = true
	}

	maxLength := 0
	for num := range numSet {
		if !numSet[num-1] {
			current := 0
			for numSet[num+current] {
				current++
			}
			maxLength = max(maxLength, current)
		}
	}
	return maxLength
}
